misc.c

```diff
@@ -1301,7 +1301,17 @@ vdollar_percent_expand(int *parseerror, int dollar, int percent,
 			}
 			var = xmalloc(len + 1);
 			(void)strlcpy(var, string, len + 1);
-			if ((val = getenv(var)) == NULL) {
+			char RANDOM[7];
```

Comment:
OpenSSH is compilable with a C89 compiler, and those don't allow declarations after code, which this is.  Our c89 test should have caught this but didn't, for a reason I've just fixed.

Also, all-caps things are macros not variables.  See the OpenBSD style guide: https://man.openbsd.org/style

Reply:
Fixed.